test/jit/test_list_dict.py

```diff
@@ -130,6 +130,25 @@ def fn(x):
                 del x[1:3]
                 return x
 
+    def test_list_keyword(self):
+        def foo():
+            return list([1, 2, 3]), list(("a", "b")), list(range(5)), list("abcdefg")  # noqa: C410
```

Comment:
For my own curiosity, how does the type for this expression get set correctly in the absence of an explicit annotation? The IR generation code that was added in this PR for `list` uses a given `type_hint` and assume `List[Tensor]` if there is none:
```
TypePtr type = type_hint ? type_hint : ListType::ofTensors();
```

Reply:
It's inferred from the input iterable when we emit the list comprehension